Projects/ChangeTier/Sources/Views/CompareTierScreen.swift

```diff
@@ -14,10 +14,31 @@ struct CompareTierScreen: View {
         vm: ChangeTierViewModel
     ) {
         self.vm = vm
+
+        let selectedQuoteId = vm.selectedQuote?.id
+
         self.limits = Dictionary(
-            uniqueKeysWithValues: vm.tiers.map({ ($0.id, $0.quotes.first?.productVariant?.insurableLimits ?? []) })
+            uniqueKeysWithValues: vm.tiers.map({
+                (
+                    $0.id,
+                    $0.quotes.first(where: { quote in
+                        quote.id == selectedQuoteId
+                    })?
+                    .productVariant?
+                    .insurableLimits ?? $0.quotes.first(where: { quote in
+                        quote.deductableAmount == vm.selectedQuote?.deductableAmount
```

Comment:
Since we have this:
```
extension Quote: Equatable {
    static public func == (lhs: Quote, rhs: Quote) -> Bool {
        return lhs.deductableAmount == rhs.deductableAmount && lhs.deductablePercentage == rhs.deductablePercentage
    }
}
```
its just enough to compare the quotes